lib/pwnlib/shellcraft/templates/amd64/setregs.rb

```diff
@@ -0,0 +1,42 @@
+require 'pwnlib/shellcraft/registers'
+require 'pwnlib/reg_sort'
+require 'pwnlib/shellcraft/shellcraft'
+
+::Pwnlib::Shellcraft.define('amd64.setregs') do |reg_context, stack_allowed: true|
+  extend ::Pwnlib::RegSort::ClassMethods
+  amd64 = ::Pwnlib::Shellcraft.instance.amd64
+  reg_context = reg_context.reject { |_, v| v.nil? }.map { |k, v| [k.to_s, v] }.to_h
+  eax = reg_context['rax']
+  edx = reg_context['rdx']
+  cdq = false
+  ev = lambda do |reg|
+    next reg unless reg.is_a?(String)
+    begin
+      evaluate(reg)
+    rescue NameError
+      reg
+    end
+  end
+  eax = ev[eax]
+  edx = ev[edx]
+
+  if eax.is_a?(Numeric) && edx.is_a?(Numeric) && eax >> 63 == edx
```

Comment:
This condition doesn't feels correct to me...
feels more like `(edx.zero? && (eax & (1 << 31)).zero?)`